robot/impl/resource_manager.go

```diff
@@ -1108,20 +1139,35 @@ func (manager *resourceManager) processResource(
 	return newRes, true, nil
 }
 
-// markResourceForUpdate marks the given resource in the graph to be updated. If it does not exist, a new node
-// is inserted. If it does exist, it's properly marked. Once this is done, all information needed to build/reconfigure
-// will be available when we call completeConfig.
-func (manager *resourceManager) markResourceForUpdate(name resource.Name, conf resource.Config, deps []string, revision string) error {
+// addToBeConstructedResource adds a new, unconfigured graph node for a resource to the
+// resource graph. If a resource with the given name already exists in the resource graph,
+// an error is returned. If that pre-existing resource is remote, it is also marked for
+// update to use the new config, as we prefer local resources to remotes in the event of
+// collision. If that pre-existing resource is local, the pre-existing resource is also
+// marked for removal, and no new graph node is created.
+func (manager *resourceManager) addToBeConstructedResource(
```

Comment:
I split `markResourceForUpdate` into two methods: `addToBeConstructedResource` and `markResourceForUpdate`. We were using `markResourceForUpdate` for both resource addition and modification, but, especially with more logic around duplicates, I think it's it worth splitting.